### PR TITLE
add methods for `vec_ptype_abbr()`

### LINKED ISSUE #379: use vec_ptype_abbr
I first noticed it here https://github.com/tidymodels/tidymodels.org/pull/269, where `bootstraps` gets shortened to `botstrps` which isn't elegant. Reprex below. This is happening because we haven't defined [vctrs::vec_ptype_abbr()](https://vctrs.r-lib.org/articles/pillar.html?q=vec_ptype_abbr#fixing-the-data-type). Easy fix, we just need to find a good abbreviation. 

``` r
library(rsample)

nested_cv(mtcars, outside = vfold_cv(), inside = bootstraps())
#> # Nested resampling:
#> #  outer: 10-fold cross-validation
#> #  inner: Bootstrap sampling
#> # A tibble: 10 × 3
#>    splits         id     inner_resamples    
#>    <list>         <chr>  <list>             
#>  1 <split [28/4]> Fold01 <botstrps [25 × 2]>
#>  2 <split [28/4]> Fold02 <botstrps [25 × 2]>
#>  3 <split [29/3]> Fold03 <botstrps [25 × 2]>
#>  4 <split [29/3]> Fold04 <botstrps [25 × 2]>
#>  5 <split [29/3]> Fold05 <botstrps [25 × 2]>
#>  6 <split [29/3]> Fold06 <botstrps [25 × 2]>
#>  7 <split [29/3]> Fold07 <botstrps [25 × 2]>
#>  8 <split [29/3]> Fold08 <botstrps [25 × 2]>
#>  9 <split [29/3]> Fold09 <botstrps [25 × 2]>
#> 10 <split [29/3]> Fold10 <botstrps [25 × 2]>

vctrs::vec_ptype_abbr(
  structure(list(), class = "bootstraps")
)
#> [1] "botstrps"

vec_ptype_abbr.bootstraps <- function(x, ...) {
  "boots"
}

nested_cv(mtcars, outside = vfold_cv(), inside = bootstraps())
#> # Nested resampling:
#> #  outer: 10-fold cross-validation
#> #  inner: Bootstrap sampling
#> # A tibble: 10 × 3
#>    splits         id     inner_resamples 
#>    <list>         <chr>  <list>          
#>  1 <split [28/4]> Fold01 <boots [25 × 2]>
#>  2 <split [28/4]> Fold02 <boots [25 × 2]>
#>  3 <split [29/3]> Fold03 <boots [25 × 2]>
#>  4 <split [29/3]> Fold04 <boots [25 × 2]>
#>  5 <split [29/3]> Fold05 <boots [25 × 2]>
#>  6 <split [29/3]> Fold06 <boots [25 × 2]>
#>  7 <split [29/3]> Fold07 <boots [25 × 2]>
#>  8 <split [29/3]> Fold08 <boots [25 × 2]>
#>  9 <split [29/3]> Fold09 <boots [25 × 2]>
#> 10 <split [29/3]> Fold10 <boots [25 × 2]>
```

<sup>Created on 2022-11-28 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by hfrick**
Closes #379

Here are some abbreviations of rsample classes - suggestions welcome! 

### Requirements

- name should be singular 
- aim: 4-6 characters
- max 8 characters?
- common prefix for "grouped", e.g. `"g_"`
- common abbr for "split", e.g. `"_splt"`
- common abbr for cross-validation rsets

### Suggestions in this PR

function | class | abbreviation | #chrs 
--- | --- | --- | ---
initial_split() | "initial_split" | `"i_splt"` | 6
initial_time_split() |  "initial_split" | `"i_splt"` | 6
group_initial_split() | "group_initial_split" | `"g_i_splt"` | 8
validation_split() | "validation_split" | `"v_splt"` | 6
validation_time_split() | "validation_time_split" | `"v_t_splt"` | 8
group_validation_split() | "group_validation_split" | `"g_v_splt"` | 8
bootstraps() | "bootstraps" | `"boot"` | 4
group_bootstraps() | "group_bootstraps" | `"g_boot"` | 6
vfold_cv() | "vfold_cv" | `"vfold"` (for consistency with grouped version) | 5
group_vfold_cv() | "group_vfold_cv" | `"g_vfold"` | 7
mc_cv() | "mc_cv" | short enough already? 
group_mc_cv() | "group_mc_cv" | `"g_mc_cv"` | 7
clustering_cv() | "clustering_cv" | `"clt_fold"` | 7
loo_cv() | "loo_cv"  | short enough already?
rolling_origin() |  "rolling_origin" | `"roll_ori"` | 8
sliding_window() | "sliding_window" | `"slide_w"` | 7
sliding_index() | "sliding_index" |  `"slide_i"` | 7
sliding_period() | "sliding_period" | `"slide_p"` | 7
nested_cv() | "nested_cv" | `"nest_cv"` | 7
apparent() | "apparent" | `"appr"` | 4
manual_rset() | "manual_rset" | `"m_rset"` | 6
permutations() | "permutations" | `"perm"` | 4

### CODE REVIEW

**Review by simonpcouch — APPROVED** [review]
These definitely seem like an improvement over defaults! No edits from me.🙂

File: R/compat-vctrs.R [hunk 1]
```diff
@@ -11,3 +11,88 @@ stop_incompatible_cast_rset <- function(x, to, ..., x_arg, to_arg) {
...
+#' @export
+vec_ptype_abbr.initial_split <- function(x, ...) {
+  "i_splt"
+}
+#' @export
+vec_ptype_abbr.group_initial_split <- function(x, ...) {
+  "g_i_splt"
+}
+
+#' @export
```
Comment (EmilHvitfeldt) [inline R/compat-vctrs.R:24]: there should be consistent spacing between functions. I like leaving 1 space between each

**Review by EmilHvitfeldt — APPROVED** [review]
I like this! 1 comment

File: R/compat-vctrs.R (hunk 1 above)
Comment (hfrick) [reply R/compat-vctrs.R:24]: ah, I was using that to group them but if that's not obvious to others then that's not a useful thing. I'll make the spacing consistent 👍

**Review by topepo — APPROVED** [review]

**Pull request merged**